applications/nrf_desktop/src/modules/config_info.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <sys/byteorder.h>
+
+#include "config_event.h"
+
+#define MODULE config_info
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_INFO_LOG_LEVEL);
+
+enum config_info_opt {
+	CONFIG_INFO_OPT_MODULE_NAME,
+	CONFIG_INFO_OPT_BOARD_NAME,
+
+	CONFIG_INFO_OPT_COUNT
+};
+
+const static char *opt_descr[] = {
+	STRINGIFY(MODULE),
+	"board_name"
+};
+
+static void fetch_config(const u8_t opt_id, u8_t *data, size_t *size)
+{
+	switch (opt_id) {
+	case CONFIG_INFO_OPT_BOARD_NAME:
+		__ASSERT_NO_MSG((strlen(CONFIG_DESKTOP_BOARD_NAME) > 0) &&
+		  (strlen(CONFIG_DESKTOP_BOARD_NAME) < CONFIG_CHANNEL_FETCHED_DATA_MAX_SIZE));
+		strcpy(data, CONFIG_DESKTOP_BOARD_NAME);
+		*size = strlen(CONFIG_DESKTOP_BOARD_NAME);
+		break;
+	default:
+		LOG_WRN("Unknown config fetch option ID %" PRIu8, opt_id);
+		break;
+	}
+}
+
+static void set_config(const u8_t opt_id, const u8_t *data, const size_t size)
+{
+	LOG_WRN("Config set is not supported");
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_module_state_event(eh)) {
+		const struct module_state_event *event =
+			cast_module_state_event(eh);
+
+		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			module_set_state(MODULE_STATE_READY);
+		}
+
+		return false;
+	}
+
+	GEN_CONFIG_EVENT_HANDLERS(opt_descr, set_config, fetch_config, true);
+
+	/* If event is unhandled, unsubscribe. */
+	__ASSERT_NO_MSG(false);
+
+	return false;
+}
+
+EVENT_LISTENER(MODULE, event_handler);
+EVENT_SUBSCRIBE(MODULE, module_state_event);
+#if CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE
+EVENT_SUBSCRIBE_FINAL(MODULE, config_event);
```

Comment:
does this module have any rationale to be compiled in when config channel is not enabled?

Reply:
yes... it is not compiled in so preprocessor checks for config channel are not needed.

Reply:
Ok, removed